internal.h

```diff
@@ -2162,6 +2162,14 @@ void rb_mutex_allow_trap(VALUE self, int val);
 VALUE rb_uninterruptible(VALUE (*b_proc)(ANYARGS), VALUE data);
 VALUE rb_mutex_owned_p(VALUE self);
 
+#define THREAD_LOCALS_INITIALISED       FL_USER13
+#define THREAD_LOCALS_INITIALISED_P(th) RB_FL_TEST_RAW((th), THREAD_LOCALS_INITIALISED)
+#define THREAD_LOCALS_INITIALISE(th)    rb_ivar_set((th), id_locals, rb_hash_new());
+#define THREAD_LOCALS_LAZY_INITIALISE(th)   if (UNLIKELY(THREAD_LOCALS_INITIALISED_P(th))) { \
```

Comment:
Shouldn't the condition be inverted?

Reply:
Correct, it was late, sorry. I will clean this up a bit to an inline function for lazy init as too many macros. Sometimes I trigger PRs with the `WIP` prefix to get ruby-core CI check the different pipelines - sorry to have taken of your time with an incomplete patch.